### PR TITLE
Fixed root grouping to consider deferred actions and placement

### PR DESCRIPTION
**Pull request by mortoray**
Fixes https://github.com/fusetools/fuselibs-public/issues/265

This PR includes the https://github.com/fusetools/fuselibs-public/pull/266 as it builds on that fix.

This PR contains:
- [x] Changelog
- [x] Documentation - minimal
- [x] Tests - incomplete, adding another one now

### CODE REVIEW

File: Source/Fuse.Triggers/Trigger.uno [hunk 1]
```diff
@@ -16,6 +16,11 @@ namespace Fuse.Triggers
 		Never,
 		/** Only changes during the rooting frame are handled as bypass, without special exceptions. */
 		Rooting,
+		
+		/** Deprecated: 2017-07-21
+			For possible backwards compatibilty, like Standard but excludes the check for layout bypass.
+			This mode should not be used. */
+		ExceptLayout,
```
Comment (kusma) [inline Source/Fuse.Triggers/Trigger.uno]: I guess "[Obsolete]" won't work on single enum values?

Comment (mortend) [inline Source/Fuse.Triggers/Trigger.uno]: I think `[Obsolete]` also works on single enum values.

Comment (mortoray) [inline Source/Fuse.Triggers/Trigger.uno]: Hmm, it does, but then using it inside Trigger also creates warnings:

`Source/Fuse.Triggers/Trigger.uno(109.36): W4139: Fuse.Triggers.TriggerBypassMode.ExceptLayout is obsolete`

How could we avoid those?

**Comment by mortoray** [comment]
I'm going to avoid merging/rebasing on master until https://github.com/fusetools/fuselibs-public/pull/266 is merged. Or I can just fold that PR into that one if we think it's okay.

File: Source/Fuse.Triggers/Trigger.uno (hunk 1 above)
Comment (kusma) [inline Source/Fuse.Triggers/Trigger.uno]: @mortoray: Ugh, yeah. How about assigning it a special value, and checking for that special value or something? Sounds a bit nasty, though...

It's not a big deal, though. We can mark this as "harder" deprecated in a follow-up if we figure out a way that doesn't suck...

**Pull request merged**